src/quack_extension.cpp

```diff
@@ -0,0 +1,63 @@
+#define DUCKDB_EXTENSION_MAIN
+
+#include "quack_extension.hpp"
+#include "duckdb.hpp"
+#include "duckdb/common/exception.hpp"
+#include "duckdb/common/string_util.hpp"
+#include "duckdb/function/scalar_function.hpp"
+
+
+#include <duckdb/parser/parsed_data/create_scalar_function_info.hpp>
+
+namespace duckdb {
+
+inline void QuackScalarFun(DataChunk &args, ExpressionState &state, Vector &result) {
+    auto &name_vector = args.data[0];
+    UnaryExecutor::Execute<string_t, string_t>(
+	    name_vector, result, args.size(),
+	    [&](string_t name) { 
+            std::string out_str = "Quack "+name.GetString()+" 🐥";
+            auto result_string = StringVector::EmptyString(result, out_str.size());
```

Comment:
This code can be simplified to:
```
return StringVector::AddString(result, "Quack "+name.GetString()+" 🐥");
```

Reply:
could be the the simpler function wasn't used back when the excel extension was implemented, i think its quite old code